src/bootstrap/compile.rs

```diff
@@ -780,9 +780,21 @@ pub fn rustc_cargo_env(builder: &Builder<'_>, cargo: &mut Cargo, target: TargetS
             cargo.env("LLVM_RUSTLLVM", "1");
         }
         let llvm_config = builder.ensure(native::Llvm { target });
-        cargo.env("LLVM_CONFIG", &llvm_config);
+
+        // HACK: on windows, paths are case insensitive, so case differences cause spurious rebuilds
+        if target.contains("windows") {
```

Comment:
Why is the *target* important here? Shouldn't it be the *host* instead? Because one might cross compile Windows from Linux, which usually happens on a case sensitive file system. Or vice versa. Also, it's an imprecise generalization that windows paths are case insensitive, because you can enable case sensitive directories on Windows. Wouldn't this break in this instance then?